core/frontend/src/render/primitives/geometry/GeometryPrimitives.ts

```diff
@@ -65,13 +65,23 @@ export abstract class Geometry {
     if (!this.displayParams.ignoreLighting) // ###TODO don't generate normals for 2d views.
       facetOptions.needNormals = true;
 
-    return this._getPolyfaces(facetOptions);
+    const polyfaceList = this._getPolyfaces(facetOptions);
+    if (undefined !== polyfaceList)

```

Comment:
Avoid wasting time applying identity transform (typical case) here and in getStrokes.